pkg/orerr/bad_request.go

```diff
@@ -0,0 +1,86 @@
+// Copyright 2022 Outreach Corporation. All Rights Reserved.
+
+// Description: Bed request error
+package orerr
+
+import (
+	"fmt"
+
+	"github.com/getoutreach/gobox/pkg/statuscodes"
+)
+
+// Violation describe particular input violation. It might be field and domain specific.
+type Violation struct {
+	// Field describes the field path with request object graph.
+	// When not provided the violation is related to a whole entitity.
+	Field *string
+
+	// Domain is optional identifier of the domain name of the validation subject.
+	Domain *string
+
+	// Reason of the error. In most of the cases the validation rule indentifier.
+	Reason string
+
+	// Additional structured details about this error. Could be used for localization of the error.
+	Metadata map[string]string
+}
+
+// NewViolation creates a new intance of Violation
+func NewViolation(reason string) *Violation {
```

Comment:
consider using value semantics, since a slice of pointers is quite cumbersome to work with later on